app/helpers/course/assessment/submission/submissions_helper.rb

```diff
@@ -13,14 +13,25 @@ def comments_container_id(submission_question)
     "course_assessment_submission_question_#{submission_question.id}_comments"
   end
 
-  # Return the last attempted answer based on the status of current submission.
-  # previous attempt if submission is in attempting state.
-  # current attempt if submission is in submitted, graded or published state.
+  # Return the last autograded attempt if the submission is being attempted,
+  # or the current_answer if it's in other states.
+  # If there are no autograded attempts, just return the last attempt.
+  #
+  # The last attempt contains the most recent autograding result if the submission is being
+  # attempted, except for questions which were answered before the `current_answer`
+  # flag was implemented.
+  # When submission is finalised, current_answer contains the autograding result.
   #
   # @return [Course::Assessment::Answer]
   def last_attempt(answer)
     submission = answer.submission
-    attempts = submission.answers.from_question(answer.question_id)
-    submission.attempting? ? attempts[-2] : attempts[-1]
+
+    attempts = submission.reload.answers.from_question(answer.question_id)
+    last_autograded_attempt = attempts.select(&:auto_grading).last
```

Comment:
Here it loads all the auto_gradings of all attempts, can we do some optimisation? This method gets called pretty frequently.